### ISSUE TITLE
Names & Perms: Ensure that it is done right from the beginning

### ISSUE THREAD

**Issue by WendyStarfall** [OpenCollar/opencollar#998]
**Version:** ``6.3.1 (160927.1)``

Licensing aside, it is important to keep the names and perms of our scripts (and other contents) as they are. Names were already shortened for 4.0 when we introduced the "new" ``oc_scriptname`` naming convention.

Our installer system relies on these names for managing script and item bundles. If a script was named differently for some reason and we wanted to run a patch, we'd end up with duplicated plugins and cores with funny numbered suffixes. Apparently some "designers" try to re-brand the stock scripts for some reason.
![facepalm](https://cloud.githubusercontent.com/assets/1100007/19930065/ea6e814a-a106-11e6-8dce-ce8734fb4f2a.png)
The user can then no longer repair the collar on her own, gets angry and ends up having to fully replace it.

While **our own distribution and Virtual Disgrace(tm) premium won't ever have this issue**, all users with 3rd party items that have those "re-branded" scripts are timebombs that go off as soon as they use our installer.

That means we get the maximum blame because someone else did something shady and made money of it:

*"Your silly update broke my collar! I paid good money for this! What do you mean talk to the seller of the item??? It says it is open collar script!" wt7z67+.2#uh<keyboard rage>8uz5hrt27+´91u6zf6gr#+´##y´232*

tl;dr :

1. keep the scripts named as they are in-world and in this repo (without the .lsl file ending of course)
2. don't mess with the permissions; opencollar works as it does because its digital rights are FULL PERM

**NOTE:** This is only relevant to stock scripts and contents! You can name plugins and apps that you made yourself in any way that works for you. Each script is its own program and if you want to restrict digital rights in some form then you can do so as well. Keep in mind that software licensing and the DRM system that we have to work with on Second Life(R) are not one and the same thing. If you want to create open source that integrates with the OpenCollar in some way, that doesn't use snippets from OpenCollar original source and that you don't want to release as full perms, you can publish your code here on the GitHub but restrict the digital rights on the "compiled" scripts as long as you inform your users where they can find the source code.
